core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -622,9 +622,9 @@ default ResolvedFunction getCoercion(Type fromType, Type toType)
     MaterializedViewFreshness getMaterializedViewFreshness(Session session, QualifiedObjectName name);
 
     /**
-     * Returns the result of redirecting the table scan on a given table to a different table.
+     * Returns the result of redirecting the table scan on the specified columns of a given table to a different table.
      * This method is used by the engine during the plan optimization phase to allow a connector to offload table scans to any other connector.
      * This method is called after security checks against the original table.
      */
-    Optional<TableScanRedirectApplicationResult> applyTableScanRedirect(Session session, TableHandle tableHandle);
+    Optional<TableScanRedirectApplicationResult> applyTableScanRedirect(Session session, TableHandle tableHandle, List<ColumnHandle> columns);
```

Comment:
We should add a new method instead that defaults to the one without column handles. 
Also mark old method deprecated.